frontend/src/components/common/useReloadPrompt.ts

```diff
@@ -0,0 +1,12 @@
+import { useEffect } from "react";
+
+const useReloadPrompt = () => {
+  useEffect(() => {
+    window.onbeforeunload = () => true;
```

Comment:
It's probably fine, but teeeeeeechnically this might not work if multiple components on the same page use this hook at once. Usually you'd prefer the `addEventListener`/`removeEventListener` API instead, but I'm fine leaving it as is since it's a bit of an edge case.

Reply:
you rite